src/components/VirtualizedGrid/VirtualizedGrid.jsx

```diff
@@ -0,0 +1,341 @@
+import React, { useRef, forwardRef, useCallback, useMemo, useEffect, useState } from "react";
+import PropTypes from "prop-types";
+import NOOP from "lodash/noop";
+import cx from "classnames";
+import { VariableSizeGrid as Grid } from "react-window";
+import AutoSizer from "react-virtualized-auto-sizer";
+import {
+  getNormalizedItems,
+  easeInOutQuint,
+  getMaxOffset,
+  getOnItemsRenderedData,
+  isVerticalScrollbarVisible
+} from "./virtualized-grid-service";
+import usePrevious from "../../hooks/usePrevious";
+import useThrottledCallback from "../../hooks/useThrottledCallback";
+import useMergeRefs from "../../hooks/useMergeRefs";
+import "./VirtualizedGrid.scss";
+
+const VirtualizedGrid = forwardRef(
+  (
+    {
+      className,
+      id,
+      items,
+      itemRenderer,
+      getRowHeight,
+      getColumnWidth,
+      onScroll,
+      getItemId,
+      scrollToId,
+      scrollDuration,
+      onScrollToFinished,
+      onItemsRendered,
+      onItemsRenderedThrottleMs,
+      onSizeUpdate,
+      onVerticalScrollbarVisiblityChange
+    },
+    ref
+  ) => {
+    // states
+    const [gridHeight, setGridHeight] = useState(0);
+    const [gridWidth, setGridWidth] = useState(0);
+
+    // prevs
+    const prevScrollToId = usePrevious(scrollToId);
+
+    // Refs
+    const componentRef = useRef(null);
+    const isVerticalScrollbarVisibleRef = useRef(null);
+    const gridRef = useRef(null);
+    const scrollTopRef = useRef(0);
+    const animationDataRef = useRef({});
+    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+
+    const animationData = animationDataRef.current;
+    if (!animationData.initialized) {
```

Comment:
pass this object in the ref initial state.- 
```javascript
useRef({
  scrollOffsetInitial: 0,
  scrollOffsetFinal: 0,
  animationStartTime: 0
})
```

Reply:
Orr it will create a new object every time (even if it won't update the ref state)

Reply:
@MosheZemah the `use` function are called